src/ConnectionAutomaton.h

```diff
@@ -22,13 +24,18 @@ using StreamId = uint32_t;
 /// Creates, registers and spins up responder for provided new stream ID and
 /// serialised frame.
 ///
-/// It is a responsibility of this startegy to register the responder with the
+/// It is a responsibility of this strategy to register the responder with the
 /// connection automaton and provide it with the initial frame if needed.
 /// Returns true if the responder has been created successfully, false if the
 /// frame cannot start a new stream, in which case the frame (passed by a
 /// mutable referece) must not be modified.
 using StreamAutomatonFactory = std::function<bool(StreamId, Payload&)>;
 
+class ConnectionCloseCallback {
```

Comment:
I was using std:function originally, switched to this because Justin explained its cheaper.  But maybe I misunderstood something.